Add end-user documentation for CCS using async-search

quux00 opened the pull request:
End user docs have been updated, with new async-search based examples in the "modules-cross-cluster-search" page, which did not have any async-search examples previously.

quux00 committed:
Change async-search CCS to use minimize_roundtrips by default.

Kibana will start using this feature once this merges.
Closes #73971

quux00 committed:
Updated search-across-clusters.asciidoc with async_seach info

internal doc tests are passing

elasticsearchmachine commented:
Pinging @elastic/es-search (Team:Search)

quux00 commented:
@abdonpijpelink I believe this is ready for a docs review. This is my first time editing end user docs with examples that are tested, so let me know where I can improve.

quux00 committed:
minor end user doc fixes

abdonpijpelink commented on docs/reference/search/search-your-data/search-across-clusters.asciidoc:
```diff
@@ -286,6 +286,321 @@ means the document came from the local cluster.
+[[ccs-async-search-minimize-roundtrips-true]]
+=== Using async search for CCS with ccs_minimize_roundtrips=true
+
+The following <<async-search,async search>> API request searches the `my-index-000001` index on
+three clusters:
```
```suggestion
Remote clusters can be queried asynchronously using the 
<<async-search,async search>> API. Async searches accept a
<<ccs-minimize-roundtrips,`ccs_minimize_roundtrips`>> parameter that defaults to
`true`. 

The following request searches the `my-index-000001` index on three clusters 
asynchronously:
```

abdonpijpelink replied:
I think it's nice to introduce the Async API and the `ccs_minimize_roundtrips` parameter before jumping into the example.

abdonpijpelink commented on docs/reference/search/async-search.asciidoc:
```diff
@@ -123,9 +123,12 @@ number of new shard responses (`5` by default).
-setting it to `true` may improve overall search latency, particularly when
-searching clusters with a large number of nodes.
+* `ccs_minimize_roundtrips` defaults to `true` (same as the synchronous search API).
+When doing a cross-cluster search, setting it to `true` may improve overall search
```
```suggestion
When doing a {ccs}, setting it to `true` may improve overall search
```

quux00 replied:
Fixed. Thanks!

abdonpijpelink reviewed (approved):
LGTM. I've left some minor suggestions.

Congrats on getting those code samples to pass the CI checks on the first attempt!

quux00 committed:
Changed docs to reflect that ccs_minimize_roundtrips defaults to false, not true.

Updated the search-across-clusters.asciidoc "Don't minimize network roundtrips" section
to reflect the fact that the REST based Search Shards API is no longer called but rather
an internal transport-layer only version of search_shards.

quux00 committed:
Changed request.setCcsMinimizeRoundtrips default for async-search back to false

javanna commented on docs/reference/search/async-search.asciidoc:
```diff
@@ -123,9 +123,12 @@ number of new shard responses (`5` by default).
-searching clusters with a large number of nodes.
+* `ccs_minimize_roundtrips` defaults to `false`. When doing a {ccs}, setting
+it to `true` may improve overall search latency, particularly when searching
+clusters with a large number of nodes. However, when set to `true`, search
```
I see number of nodes being mentioned, but like I mentioned before, the performance is a factor of the number of shards and not the number nodes.

javanna replied:
ops, seems like I had started a review which I forgot to submit, no wonder why the comment was not addressed :)

quux00 replied:
Thanks. Fixed.

javanna reviewed (approved):
LGTM besides the small comment I left

quux00 committed:
PR feedback
